### PR TITLE
Add failure cases to execution

### PR DESCRIPTION
**Pull request by krl**

### CODE REVIEW

File: vmx/tests/fallible.rs [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (miloszm) [inline vmx/tests/fallible.rs:1]: Is this file empty?

**Review by miloszm — CHANGES_REQUESTED** [review]

File: piecrust-uplink/src/state.rs [hunk 2]
```diff
@@ -253,8 +262,8 @@ impl<S> State<S> {
         &mut self,
         mod_id: ModuleId,
         raw: RawTransaction,
-    ) -> RawResult {
-        // Necessary to avoid ruling out potential memory changes from recursive
+    ) -> Result<RawResult, ModuleError> {
+        // Neccesary to avoid ruling out potential memory changes from recursive
```
Comment (miloszm) [inline piecrust-uplink/src/state.rs:266]: nitpick: spelling error in 'necessary'

**Review by miloszm — APPROVED** [review]
LGTM, one nitpick

**Review by ureeves — APPROVED** [review]
LGTM

File: piecrust-uplink/src/state.rs (hunk 2 above)
Comment (krl) [reply piecrust-uplink/src/state.rs]: fixed

**Pull request merged**